File: BOJ/yein/2_week/차집합.py
na, nb = map(int, input().split())
a = list(map(int, input().split()))
b = list(map(int, input().split()))

# 시간초과
# arr = []
# for aa in a:
#     if (aa not in b):
#         arr.append(aa)

# 시간초과 22
# arr = [x for x in a if x not in b]

arr = set(a)-set(b)

print(len(arr))
if (len(arr)!=0):
    for i in sorted(arr):
        print(i, end=" ")

File: BOJ/jaehong/2_week/차집합.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.HashSet;

public class 차집합 {
  public static void main(String[] args) throws IOException {
    BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));

    String[] ABSize = bufferedReader.readLine().split(" ");
    String[] setStrA = bufferedReader.readLine().split(" ");
    String[] setStrB = bufferedReader.readLine().split(" ");

    HashSet<Integer> setNumA = new HashSet<>();
    HashSet<Integer> setNumB = new HashSet<>();

    for (String num : setStrA) {
      setNumA.add(Integer.parseInt(num));
    }

    for (String num : setStrB) {
      setNumB.add(Integer.parseInt(num));
    }

    setNumA.removeAll(setNumB); // 핵심

    System.out.println(setNumA.size());
    setNumA.stream().sorted().forEach(num -> System.out.print(num + " "));
  }
}
